core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/authorization/JDBCAuthorizationManager.java

```diff
@@ -801,7 +801,14 @@ private  void addAuthorizationForRole(String roleName, String resourceId, String
                 }
             }
             dbConnection.commit();
-        } catch (SQLException e) {
+        } catch (Exception e) {
```

Comment:
What are we catching Exception here ?. Please catch only the relevant Exceptions and update the PR.